app/code/Magento/ConfigurableProductGraphQl/Model/Cart/BuyRequest/SuperAttributeDataProvider.php

```diff
@@ -76,6 +77,10 @@ public function execute(array $cartItemData): array
         } catch (NoSuchEntityException $e) {
             throw new GraphQlNoSuchEntityException(__('Could not find specified product.'));
         }
+        $configurableProductLinks = $parentProduct->getExtensionAttributes()->getConfigurableProductLinks();
+        if (!in_array($product->getId(), $configurableProductLinks)) {
+            throw new GraphQlInputException(__('The child product do not belong to the parent product.'));
```

Comment:
I'd prefer to not to expose redundant info and say something like `Could not find specified product.`